storage/provider/loop.go

```diff
@@ -66,16 +90,69 @@ func (lvs *loopVolumeSource) rootDeviceDir() string {
 	return filepath.Join(dirParts...)
 }
 
-func (lvs *loopVolumeSource) CreateVolumes(params []storage.VolumeParams) ([]storage.BlockDevice, error) {
-	panic("not implemented")
+func (lvs *loopVolumeSource) CreateVolumes(args []storage.VolumeParams) ([]storage.BlockDevice, error) {
+
+	blockDevices := make([]storage.BlockDevice, len(args))
+	for idx, arg := range args {
+		id := providerId(arg)
+		filePath := filepath.Join(lvs.rootDeviceDir(), id)
+		if err := createBlockFile(lvs.runCmd, filePath, arg.Size); err != nil {
+			return nil, err
+		}
+
+		deviceName, err := attachToBlockDevice(lvs.runCmd, filePath)
+
+		// If we ran out of loop devices, create another.
+		if err == NoLoopDeviceErr {
+			nextDevNum, err := findAvailableDeviceNumber(lvs.runCmd)
+			if err == nil {
+				createLoopDevice(lvs.runCmd, nextDevNum)
+				deviceName, err = attachToBlockDevice(lvs.runCmd, filePath)
+			}
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		blockDevice := storage.BlockDevice{
+			Name:       arg.Name,
+			ProviderId: id,
+			DeviceName: deviceName,
+			Size:       arg.Size,
+			InUse:      false,
+		}
+		blockDevices[idx] = blockDevice
+		lvs.volIdToBlockDevice[id] = blockDevicePlus{blockDevice, filePath}
+	}
+
+	return blockDevices, nil
 }
 
 func (lvs *loopVolumeSource) DescribeVolumes(volIds []string) ([]storage.BlockDevice, error) {
-	panic("not implemented")
+
+	blockDevices := make([]storage.BlockDevice, len(volIds))
+	for idx, volId := range volIds {
+		device, ok := lvs.volIdToBlockDevice[volId]
```

Comment:
I've left it this way for now since I'd have to write some code to inspect the loop devices to transform them back into the BlockDevice struct. I'll take care of this later.